Readme.md

```diff
@@ -0,0 +1,171 @@
+
+This functionality is provided by workflows with the syntax `cdelivery-*.yml`. When triggered, two jobs are run: build-deploy and test. 
+
+### Continuous Deployment
+When the code is considered to be production-ready, it can be automatically deployed to the production environment.
+
+This functionality is provided by workflows with the syntax `cdeployment-*.yml`. When triggered, one job is run: build-deploy. 
+
+## How to
+### Adding a caller workflow
+Whenever you want to add a specific workflow to your repository, you should copy the appropriate caller template from `caller-workflow-templates` folder in the `.github/workflows` folder of your repository. The template should match the distribution/delivery strategy that your repository uses (AWS S3 or AWS ECS).
+
+Then, the required inputs and secrets should be included/adapted based on the needs of the repository. 
+
+### Triggering a workflow
+Workflow triggers are events that cause a workflow to run. All the Continuous Integration workflows can be manually run from the Actions tab of your repository. 
+
+![Action tab](assets/images/learn-github-actions-repository.png)
+
+Write access to the repository is required to perform these steps.
+
+1. In the left sidebar, click the workflow you want to run.
+
+    ![Action select](assets/images/actions-select-workflow.png)
+
+2. Above the list of workflow runs, select Run workflow.
+
+    ![Action dispatch](assets/images/actions-workflow-dispatch.png)
+
+3. Use the Branch dropdown to select the workflow's branch. Click Run workflow.
+
+    ![Action run](assets/images/actions-manually-run-workflow.png)
+
+### Viewing a workflow
+
+After a workflow run has started, you can see a visualization graph of the run's progress and view each step's activity on GitHub.
+
+1. In the left sidebar, click the workflow you want to see.
+
+    ![Workflow](assets/images/learn-github-actions-workflow.png)
+
+2. Under "Workflow runs", click the name of the run you want to see.
+
+    ![Workflow run](assets/images/learn-github-actions-run.png)
+
+3. Under Jobs or in the visualization graph, click the job you want to see.
+
+    ![Job result](assets/images/overview-actions-result-navigate.png)
+
+4. View the results of each step.
+
+    ![Step result](assets/images/overview-actions-result-updated-2.png)
+
+### Updating environment variables (if needed)
+The environment variables are passed to the caller workflows as either inputs or secrets. 
+Whenever you want to add a new environment variable to your repository and pass it as a secret, you should update your workflow as well. 
+
+For this purpose, you need to perform the following steps. 
+#### 1. Update inputs and secrets in a reusable workflow definition
+```
+on:
+  workflow_call:
+    inputs:
+      input-variable:
+        required: true
+        type: string
+    secrets:
+      secret-variable:
+        required: true
+```
```

Comment:
This workflow is in graasp-deploy right? (precise)
If yes, then would it be possible to add a new variable without changing the reusable workflow? For instance a variable which will be used only in one repo (so no need to refer it in the reusable workflow).

Reply:
I think your example is too generic, maybe provide an example file name if we were to change a variable for let's say a `app-name` frontend.

And is it the same procedure for backend/ECS deployment?

Reply:
> If yes, then would it be possible to add a new variable without changing the reusable workflow? For instance a variable which will be used only in one repo (so no need to refer it in the reusable workflow).
It is not possible to add a new variable without changing the reusable workflow. The advantage of reusable workflows is that it performs all the steps of the pipeline and it controls all the variables. 
- Without reusable workflows, each repo would have its own workflow, steps, variables... and everything could become a mess because you'd lose count of which repo has which variables. 
- With reusable workflows, all the variables are controlled from Graasp-deploy, and you just pass the ones your repo needs. Therefore to add a variable you add it to the reusable workflow and to your repo, but you don't have to modify the rest.

Reply:
Yes, this is possible to do in a normal workflow, but not in a reusable one (the one with `uses` syntax).